lib/manager.js

```diff
@@ -286,25 +284,24 @@ internals.Manager.prototype.registerHelper = function (name, helper) {
 };
 
 
-internals.Manager.prototype.render = function (filename, context, options, callback) {
-
-    if (!callback) {
-        return internals._wrapMethod(this, this.render, [filename, context, options]);
-    }
-
-    this._prepare(filename, options, (err, compiled) => {
+internals.Manager.prototype.render = function (filename, context, options, request) {
 
-        if (err) {
-            return callback(err);
-        }
+    return new Promise((resolve, reject) => {
 
-        this._render(compiled, context, null, (err, rendered) => {
+        this._prepare(filename, options, (err, compiled) => {
 
             if (err) {
-                return callback(err);
+                return reject(err);
             }
 
-            return callback(null, rendered, compiled.settings);
+            this._render(compiled, context, request, (err, rendered) => {
+
+                if (err) {
+                    return reject(err);
+                }
+
+                return resolve(rendered);
+            });
```

Comment:
This was the only function using `_wrapMethod` in the manager. I'm thinking about moving away from callbacks in this file, replacing with async/await